### PR TITLE
[Tests-Only] add api tests for checking etag propagation on copy operations

### LINKED ISSUE #251: [OCIS-Storage] create etag-propagation tests for copy operations
copy operations are not tested yet

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add API acceptance tests for etag propagation on copying operations

## Related Issue

- Fixes https://github.com/owncloud/product/issues/251

## How Has This Been Tested?

:robot:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiWebdavEtagPropagation2/copyFileFolder.feature [hunk 1]
```diff
@@ -0,0 +1,188 @@
+@api @skipOnOcis-OC-Storage
+Feature: propagation of etags when copying files or folders
+
+  Background:
+    Given user "Alice" has been created with default attributes and without skeleton files
+
+  Scenario Outline: coping a file does not changes its etag
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavEtagPropagation2/copyFileFolder.feature:7]:
```suggestion
  Scenario Outline: copying a file does not change its etag
```

File: tests/acceptance/features/apiWebdavEtagPropagation2/copyFileFolder.feature [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    Given using <dav_version> DAV path
+    And user "Alice" has uploaded file with content "uploaded content" to "file.txt"
+    And user "Alice" has stored etag of element "/file.txt"
+    And user "Alice" has stored etag of element "/file.txt" on path "/renamedFile.txt"
+    When user "Alice" copies file "/file.txt" to "/renamedFile.txt" using the WebDAV API
+    Then these etags should not have changed:
+      | user  | path      |
+      | Alice | /file.txt |
+    And these etags should have changed:
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavEtagPropagation2/copyFileFolder.feature:16]: We could also store the etag of `/` and check that it changes after `renamedFile.txt` gets created.

File: tests/acceptance/features/apiWebdavEtagPropagation2/copyFileFolder.feature [hunk 3]
```diff
@@ -0,0 +1,188 @@
+@api @skipOnOcis-OC-Storage
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdavEtagPropagation2/copyFileFolder.feature:1]: What to do about skipping these on OC storage?
Are the other etag tests skipped on OC storage?

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/27451/2/7

It failed coding-style @dpakach

File: tests/acceptance/features/apiWebdavEtagPropagation2/copyFileFolder.feature (hunk 3 above)
Comment (dpakach) [reply tests/acceptance/features/apiWebdavEtagPropagation2/copyFileFolder.feature:1]: Yes all the etag tests are skipped in oc storage currently. I think etags propagation currently doesn't works in oc storage, We will have to run and update the expected failure file for ocis storage when updating commit id there.

Comment (phil-davis) [reply tests/acceptance/features/apiWebdavEtagPropagation2/copyFileFolder.feature:1]: OK - removing these sort of tags is a separate issue

**Review by phil-davis — APPROVED** [review]

**Pull request merged**